### PR TITLE
ENH: Allow plotting backend to be an option

### LINKED ISSUE #28619: Allow users to pass backend option to `.plot()`
Once you have installed another plotting library with a pandas.plotting entrypoint it would be nice to be able to toggle back and forth between different backends even within on session.

## What currently exists
```python
pd.options.plotting.backend = 'holoviews'
```

## Proposed additional API

```python
df.plot(backend='holoviews')

```

This would be particularly useful when a particular backend doesn't support a certain type of plot, or you need to use a backend to get some specific functionality but don't want to use if for your whole notebook.
 
@datapythonista - @TomAugspurger mentioned that you might have opinions on this.

### PR DESCRIPTION
**Pull request by jsignell**
- [x] closes #28619
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: doc/source/whatsnew/v1.0.0.rst [hunk 1]
```diff
@@ -265,6 +265,7 @@ Plotting
 - Bug where :meth:`DataFrame.boxplot` would not accept a `color` parameter like `DataFrame.plot.box` (:issue:`26214`)
 - Bug in the ``xticks`` argument being ignored for :meth:`DataFrame.plot.bar` (:issue:`14119`)
 - :func:`set_option` now validates that the plot backend provided to ``'plotting.backend'`` implements the backend when the option is set, rather than when a plot is created (:issue:`28163`)
+- :meth:`.plot` now allow a ``backend`` kwarg to allow changing between backends in one session (:issue:`28619`).
```
Comment (TomAugspurger) [inline doc/source/whatsnew/v1.0.0.rst]:
```suggestion
- :meth:`DataFrame.plot` now allow a ``backend`` keyword arugment to allow changing between backends in one session (:issue:`28619`).
```

**Comment by jsignell** [comment]
Hmm I haven't added any docs. I suppose there should at least be a mention in the docstring.

File: pandas/plotting/_core.py [hunk 2]
```diff
@@ -373,7 +373,7 @@ def boxplot(
         >>> type(boxplot)
         <class 'numpy.ndarray'>
     """
-    plot_backend = _get_plot_backend("matplotlib")
+    plot_backend = _get_plot_backend(kwds.pop("backend", None))
```
Comment (TomAugspurger) [inline pandas/plotting/_core.py]: cc @datapythonista on this one. Do you recall why "matplotlib" was hardcoded here? Wasn't part of the backend API?

**Review by TomAugspurger — COMMENTED** [review]
Can you also add `backend` to the docstring (and possibly the signature) for all of the plotting methods? That'll make this a bit more discoverable.

File: pandas/plotting/_core.py (hunk 2 above)
Comment (datapythonista) [reply pandas/plotting/_core.py]: I"m on my phone and I can't check now, but I think this is the boxplot that is independent from the accessor, `pandas.plotting.boxplot`. There is a proposal to deprecate it, but regardless of that the decision was to only switch the backend for the accessor plots.

Comment (jsignell) [reply pandas/plotting/_core.py]: It was really unclear to me why the matplotlib was hardcoded there. The function seems very similar to `boxplot_frame` and `boxplot_frame_groupby`.

Comment (datapythonista) [reply pandas/plotting/_core.py]: #28177 can provide some more context

Comment (jsignell) [reply pandas/plotting/_core.py]: I understand that the API is being unified and all of these boxplot* are being deprecated, which seems like a great plan. I just don't get why the backend should be hardcoded when calling `pandas.plotting.boxplot` and not when calling `df.boxplot` it seems oddly inconsistent.

Comment (jsignell) [reply pandas/plotting/_core.py]: Tom pointed me to https://github.com/pandas-dev/pandas/pull/28159, so I have plenty of explanation now. Feel free to ignore my last comment :)

**Comment by jsignell** [comment]
I think this is ready.

**Comment by jbrockmendel** [comment]
@jsignell can you rebase

File: pandas/plotting/_core.py [hunk 3]
```diff
@@ -395,9 +404,177 @@ def boxplot_frame(
...
+        <class 'pandas.core.series.Series'>
+
+    If ``return_type`` is `None`, a NumPy array of axes with the same shape
+    as ``layout`` is returned:
+
+        >>> boxplot = df.boxplot(column=['Col1', 'Col2'], by='X',
+        ...                      return_type=None)
+        >>> type(boxplot)
+        <class 'numpy.ndarray'>
+    """
```
Comment (jsignell) [inline pandas/plotting/_core.py]: I'd like not to duplicate this, but I couldn't figure out a good way to inject the backend argument doc string.

**Review by TomAugspurger — APPROVED** [review]

**Review by datapythonista — APPROVED** [review]
lgtm, thanks @jsignell

**Pull request merged**